script.js

```diff
@@ -1,55 +1,63 @@
-function Animal(name){
+class Animal {
+  constructor(name, legs, img) {
     this.name = name;
-}
+    this.legs = legs;
+    this.img = img;
+  }
 
-Animal.prototype.action = function(){
-  document.getElementById(this.actionSoundName).play();
-  
-}
+  action() {
+    document.getElementById(this.actionSoundName).play();
+  }
```

Comment:
`   action(e) {
        e.stopPropagation();
        document.getElementById(this.actionSoundName).play();
    }`
    Şu şekilde `e.stopPropagation` ekleyerek Event Bubbling'e engel olabilirsin. Böylece button'un on click'i tr'nin onclick'ini tetiklemez.